### PR TITLE
Add calico version check, separate cni version check

### PR DESCRIPTION
**Pull request by mdrahman-suse**
#### Proposed Changes ####

 
Separating tests for CNIs

No update to docs needed

#### Testing ####

Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.

2. Verify code lint; we should not have errors.

3. Update the documentation if needed.

4. Update makefile and docker run if adding new tests.

5. Run your tests at least 4 times with all configurations needed and possible.

6. If needed test with different os types.

### CODE REVIEW

**Review by aganesh-suse — DISMISSED** [review]
LGTM

**Comment by fmoral2** [comment]
can we please update the bump docs?

File: pkg/customflag/validate.go [hunk 1]
```diff
@@ -36,7 +36,11 @@ func ValidateTemplateFlags() {
 	case "components":
 		validateComponentsTest(expectedValues, expectedUpgrades)
 	case "flannel":
```
Comment (fmoral2) [inline pkg/customflag/validate.go:38]: we need to update here as well with the new tags 
https://github.com/rancher/distros-test-framework/blob/7eb69146098fe68ee67a80a1f51522dbd29391a0/pkg/customflag/validatejenkins.go#L80

File: entrypoint/versionbump/cniflannel_test.go [hunk 2]
```diff
@@ -33,16 +35,16 @@ var _ = Describe("Flannel Version bump:", func() {
 	})
 
 	It("Test flannel version bump", func() {
-		cmd := "kubectl get node -o yaml : | grep 'hardened-flannel' -A1"
+		flannel := "kubectl get node -o yaml : | grep 'hardened-flannel' -A1"
```
Comment (ShylajaDevadiga) [inline entrypoint/versionbump/cniflannel_test.go:38]: Can we call it flannelCmd as these are commands to retrieve version? Similarly above ciliumCmd and cniPluginsCmd

Comment (mdrahman-suse) [reply entrypoint/versionbump/cniflannel_test.go]: Updated

File: pkg/customflag/validate.go (hunk 1 above)
Comment (mdrahman-suse) [reply pkg/customflag/validate.go:38]: Updated

**Review by fmoral2 — DISMISSED** [review]

**Review by aganesh-suse — DISMISSED** [review]
LGTM

File: pkg/customflag/validate.go [hunk 3]
```diff
@@ -89,7 +93,8 @@ func validateTestTagFromLocal() string {
 	}
 
 	if testTag != "versionbump" && testTag != "cilium" && testTag != "multus" &&
-		testTag != "components" && testTag != "flannel" {
+		testTag != "components" && testTag != "flannel" && testTag != "calico" &&
+		testTag != "canal" {
```
[the remark this answers is not in the document]
Comment (mdrahman-suse) [reply pkg/customflag/validate.go]: Thanks! Updated

**Comment by mdrahman-suse** [comment]
Please review as some refactor went in

File: pkg/customflag/validate.go [hunk 4]
```diff
@@ -125,6 +130,35 @@ func validateCiliumTest(expectedValue, valuesUpgrade []string) {
 	}
 }
 
+func validateCanalTest(expectedValue, valuesUpgrade []string) {
+	// calico + flannel
+	canalCmdsLength := 2
```
[the remark this answers is not in the document]
Comment (mdrahman-suse) [reply pkg/customflag/validate.go]: I have optimized the code in a way that it is not needed anymore

**Review by VestigeJ — APPROVED** [review]

**Comment by mdrahman-suse** [comment]
Updated please check @ShylajaDevadiga @fmoral2 @VestigeJ

**Review by VestigeJ — APPROVED** [review]

**Pull request merged**